starknet_py/net/gateway_client.py

```diff
@@ -285,12 +295,21 @@ async def get_class_by_hash(self, class_hash: Hash) -> DeclaredContract:
         return DeclaredContractSchema().load(res, unknown=EXCLUDE)
 
     async def _add_transaction(
-        self, tx: StarknetTransaction
+        self,
+        tx: StarknetTransaction,
+        token: Optional[str] = None,
     ) -> SentTransactionResponse:
-        res = await self._gateway_client.post(
-            method_name="add_transaction",
-            payload=StarknetTransaction.Schema().dump(obj=tx),
-        )
```

Comment:
How about `params = {"token": token} if token is not None else {}` then you don't have to duplicate client.post(...) code.